### PR TITLE
New Staff UI to add up to three topics to Activities

### PR DESCRIPTION
**Pull request by cissyxyu**
## WHAT
Created a new React UI for staff to add up to three level one topics to Activities.

## WHY
We want staff to be able to set topics on activities and create new topics.

## HOW
Create a React class TopicGroup that contains all 3 selector elements you would need to select a new level one topic (the level 3 topic selector, level 2 topic selector, and level 1 topic selector). Make the parent class Topic create three of these TopicGroups instead of three TopicColumns like it was doing before.

### Screenshots
Screen Shot 2022-09-28 at 3 50 46 PM

### Notion Card Links
https://www.notion.so/quill/Activity-UI-Supports-3-tags-163deee555324ebbb7baa1b6e08a8af3

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  YES
Have you deployed to Staging? | YES
Self-Review: Have you done an initial self-review of the code below on Github? | Yes
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | Yes

### CODE REVIEW

File: services/QuillLMS/client/app/bundles/Staff/components/activityForm/topicColumn.tsx
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Staff/components/activityForm/topicColumn.tsx:65]: You shouldn't need an `onKeyDown` for a `button` element (it will automatically be handled by the `onClick`, but that isn't true for interactive elements).

File: services/QuillLMS/client/app/bundles/Staff/components/activityForm/topicColumn.tsx
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Staff/components/activityForm/topicColumn.tsx:61]: This doesn't get reset so it should just be a const.

File: services/QuillLMS/client/app/bundles/Staff/components/activityForm/topicColumn.tsx
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Staff/components/activityForm/topicColumn.tsx]: For code cleanliness I would probably pull this out into a constant instead of doing the iteration inside of the render.

File: services/QuillLMS/client/app/bundles/Staff/components/activityForm/topicColumn.tsx
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Staff/components/activityForm/topicColumn.tsx]: You shouldn't need an onKeyDown for a button element (it will automatically be handled by the onClick, but that isn't true for interactive elements).

File: services/QuillLMS/client/app/bundles/Staff/components/activityForm/topicColumn.tsx
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Staff/components/activityForm/topicColumn.tsx]: This doesn't get reset so it should just be a const.

File: services/QuillLMS/client/app/bundles/Staff/components/activityForm/topicGroup.tsx
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Staff/components/activityForm/topicGroup.tsx]: Should this be `levelTwoTopic.id ? ...` ?

File: services/QuillLMS/client/app/bundles/Staff/components/activityForm/topicGroup.tsx
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Staff/components/activityForm/topicGroup.tsx]: It feels like a little bit of a code smell to explicitly call `handleLevelOneChange` before you set it when we could use an effect hook to do that when the value changes. That's a little complicated by the fact that you need the old value to pass up to the function, but you could do something like [this](https://stackoverflow.com/questions/53446020/how-to-compare-oldvalues-and-newvalues-on-react-hooks-useeffect) which I think would still be cleaner. Up to you, just wanted to flag it as a little unparadigmatic.

File: services/QuillLMS/client/app/bundles/Staff/components/activityForm/topicGroup.tsx
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Staff/components/activityForm/topicGroup.tsx]: Since this `if` block basically gets repeated in the `removeTopic` function below, I wonder if it's worth setting up a const like:

```
const levelNumberToFunction = {
  1: setLevelOneTopic,
  2: setLevelTwoTopic,
  3: setLevelThreeTopic
}
```

And then here you could just call `levelNumberToTopic[level](newTopic)` and below you would call `levelNumberToTopic[level]({})`.

File: services/QuillLMS/client/app/bundles/Staff/components/activityForm/topics.tsx
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Staff/components/activityForm/topics.tsx]: Let's import this into a style file rather than into the component itself - that isn't really a pattern we use elsewhere (arguably it's one we should move towards, but I think better to be consistent for now).

**Review by emilia-friedberg — APPROVED** [review]
Mostly nits - feel free to re-request a review if you want eyes on it again but otherwise LGTM once those changes are made. Nice work!

**Review by happythenewsad — APPROVED** [review]

**Review by emilia-friedberg — APPROVED** [review]

**Comment by emilia-friedberg** [comment]
Thanks for making those changes! Looks great.

**Review by brendanshean — APPROVED** [review]
This looks very cool!

**Pull request merged**
